File: Java/NetBeans/QLTV/src/qltv/Sach.java
package qltv;

public class Sach extends TaiLieu{
    protected String tacGia;
    protected String nhaXB;
    protected int namXB;
    
    public Sach(String maTL,
                String tenTL,
                int viTri,
                String tacGia,
                String nhaXB,
                int namXB) {
        super(maTL, tenTL, viTri);
        this.tacGia = tacGia;
        this.nhaXB = nhaXB;
        this.namXB = namXB;
    }
    
    public void hienThi() {
        super.hienThi();
        System.out.println("Tac gia: " + tacGia);
        System.out.println("Nha xuat ban: " + nhaXB);
        System.out.println("Nam xuat ban: " + namXB);
    }
}

File: C#/QLThuVien/QLThuVien/Sach.cs
namespace QLThuVien
{
    class Sach : TaiLieu
    {
        protected string tacGia;
        protected string nhaXB;
        protected int namXB;

        public string TacGia { get => tacGia; set => tacGia = value; }

        public Sach(string maTL, string tenTL, int viTri, string tacGia, string nhaXB, int namXB) : base(maTL, tenTL, viTri)
        {
            this.tacGia = tacGia;
            this.nhaXB = nhaXB;
            this.namXB = namXB;
        }

        public override void hienThi()
        {
            base.hienThi();
            Console.WriteLine($"Tac gia: {tacGia}");
            Console.WriteLine($"Nha xuat ban: {nhaXB}");
            Console.WriteLine($"Nam xuat ban: {namXB}");
        }

        public override string loaiTL()
        {
            return "Sach";
        }
    }
}